translate/v23tov30/v23tov30.go

```diff
@@ -106,6 +106,24 @@ func Check2_3(cfg old.Config, fsMap map[string]string) error {
 			}
 		}
 	}
+
+	// check that there are no duplicates with systemd units or dropins
+	unitMap := map[string]struct{}{}  // unit name -> struct{}
```

Comment:
nit: I'd personally find it easier to read if this were a map to bools.
It would also allow for shortening the conditional to just  `if unitMap[unit.Name] {`

Not a blocker, and :+1: otherwise :)